### PR TITLE
refactor: Allow evaluate variables for dsn and uss path

### PR DESCRIPTION
**Pull request by Nurkambay**

### CODE REVIEW

**Review by temanbrcom — APPROVED** [review]
🍅

File: clients/cobol-lsp-vscode-extension/src/services/Settings.ts [hunk 1]
```diff
@@ -119,7 +113,7 @@ export class SettingsService {
     private static evaluateVariable(dataList: string[], variable: string, value: string): string[] {
         const result: string[] = [];
         if (dataList) {
-            dataList.forEach(d => result.push(d.replace("$" + variable, value)))
+            dataList.forEach(d => result.push(d.replace(`$\{${variable}}`, value)))
```
Comment (grianbrcom) [inline clients/cobol-lsp-vscode-extension/src/services/Settings.ts:116]: Seems that you need to escape the second brace like this: `$\{${variable}\}`.
But it works as it is...

**Review by grianbrcom — APPROVED** [review]
:butterfly:

File: clients/cobol-lsp-vscode-extension/src/__tests__/services/SettingsTest.spec.ts [hunk 2]
```diff
@@ -78,23 +78,23 @@ describe("Validate URI generation for a given workspace folder", () => {
 describe("SettingsService evaluate variables", () => {
     test("Evaluate program_file", () => {
         vscode.workspace.getConfiguration = jest.fn().mockReturnValue({
-            get: jest.fn().mockReturnValue(["copybook/$program_file"]),
+            get: jest.fn().mockReturnValue(["copybook/${program_file}"]),
```
[the remark this answers is not in the document]
Comment (Nurkambay) [reply clients/cobol-lsp-vscode-extension/src/__tests__/services/SettingsTest.spec.ts:81]: It is described in the user story I was working on

File: clients/cobol-lsp-vscode-extension/src/services/Settings.ts (hunk 1 above)
Comment (Nurkambay) [reply clients/cobol-lsp-vscode-extension/src/services/Settings.ts]: Fixed, anyway

**Review by grianbrcom — CHANGES_REQUESTED** [review]
You've commented your `test-report.xml`. It's from IDEA. Can you add it into `.gitignore`?

**Comment by Nurkambay** [comment]
> You've commented your `test-report.xml`. It's from IDEA. Can you add it into `.gitignore`?

Yep, thanks

**Review by temanbrcom — APPROVED** [review]

**Pull request merged**